redaxo/src/core/lib/util/socket/socket_response.php

```diff
@@ -179,23 +204,6 @@ public function getBufferedBody($length = 1024)
         if (feof($this->stream)) {
             return false;
         }
-        if ($this->chunked) {
-            if (0 == $this->chunkPos) {
-                $this->chunkLength = hexdec(fgets($this->stream));
-                if (0 == $this->chunkLength) {
-                    return false;
-                }
-            }
-            $pos = ftell($this->stream);
-            $buf = fread($this->stream, min($length, $this->chunkLength - $this->chunkPos));
-            $this->chunkPos += ftell($this->stream) - $pos;
-            if ($this->chunkPos >= $this->chunkLength) {
-                fgets($this->stream);
-                $this->chunkPos = 0;
-                $this->chunkLength = 0;
-            }
-            return $buf;
-        }
```

Comment:
getBufferedBody ist public. Also es muss funktionieren, wenn jemand den Body selbst per getBufferedBody abruft, ganz ohne getBody oder writeBodyTo aufzurufen.

Also dieser Code muss so eigenständig funktionieren:

```php
$socket = rex_socket::factoryUrl($url);
$response = $socket->doGet();

while (false !== ($buf = $response->getBufferedBody())) {
    // do something with $buf
}
```

Sehe ich das richtig, dass er nach deiner Änderung nicht mehr funktionieren würde wenn chunked?
Im Optimalfall würde auch bei der Variante der Body dekodiert werden, wenn die Option aktiivert wurde.

Reply:
Das stimmt wohl. Theoretisch müsste es funktionieren, das ganze Stream Filter Handling direkt in die getBufferedBody Methode zu implementieren. Dann müssen die Filter allerdings als properties der Klasse definiert werden, damit man in der Methode prüfen kann ob diese bereits hinzugefügt wurden oder nicht.

Werde da etwas bauen.